presto-hive/src/main/java/com/facebook/presto/hive/authentication/KerberosAuthentication.java

```diff
@@ -102,4 +103,12 @@ private static Configuration createConfiguration(String principal, String keytab
             }
         };
     }
+
+    @VisibleForTesting
+    public KerberosAuthentication(String principal)
```

Comment:
make package private and put below the other constructor

or, better, remove, because the object creating with this constructor seems effectively unused

Reply:
Hi @findepi , thanks for your suggestions and comments. 

Creating KerberosAuthentication since we need a KerberosHadoopAuthentication object for setting configurations. (The constructor for KerberosHadoopAuthentication requires KerberosAuthentication to be non-null.) 

If we don't want to create KerberosAuthentication at all, will it be okay to just call the relavant functions (validateHDFSConfiguration and UserGroupInformation::setConfiguration) directly from the Test?

Reply:
IMO it would be better. However, see a comment about tests -- since this functionality is static fields-based, tests are not effectively isolated, even if you do the cleanup at the end.
I think it would be better to have a product test (in presto, product tests are tests running actual systems with docker, emulating distributed setup)